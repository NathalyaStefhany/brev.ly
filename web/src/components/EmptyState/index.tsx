import React, { ComponentProps, ReactElement } from 'react';

type EmptyStateProps = {
  icon: ReactElement;
  description: string;
} & ComponentProps<'div'>;

export const EmptyState: React.FC<EmptyStateProps> = ({
  icon,
  description,
  ...props
}) => {
  return (
    <div
      className="flex flex-col items-center justify-center gap-6 mt-16 mb-12"
      {...props}
    >
      {icon}

      <p className="text-xs text-gray-500 uppercase text-center">
        {description}
      </p>
    </div>
  );
};
