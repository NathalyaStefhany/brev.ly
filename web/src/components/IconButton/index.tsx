import React, { ComponentProps, ComponentType } from 'react';

import { IconProps } from '@phosphor-icons/react';

type IconButtonProps = ComponentProps<'button'> & {
  icon: ComponentType<IconProps>;
};

export const IconButton: React.FC<IconButtonProps> = ({
  icon: Icon,
  ...props
}) => {
  return (
    <button
      type="button"
      className="cursor-pointer transition-all duration-300 bg-gray-200 border border-gray-200 rounded-sm p-3.5 hover:not-disabled:border-blue-base disabled:opacity-50 disabled:cursor-not-allowed"
      {...props}
    >
      <Icon size="1rem" color="var(--color-gray-600)" />
    </button>
  );
};
