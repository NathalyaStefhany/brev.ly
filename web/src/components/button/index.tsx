import React, { ComponentProps, ComponentType } from 'react';

import { tv, VariantProps } from 'tailwind-variants';
import { IconProps } from '@phosphor-icons/react';

const buttonVariants = tv({
  base: 'flex flex-row gap-3 items-center justify-center cursor-pointer transition-all duration-300 disabled:cursor-not-allowed',
  variants: {
    variant: {
      primary:
        'max-w-176 w-full bg-blue-base text-md text-white rounded-lg px-10 py-7.5 hover:not-disabled:bg-blue-dark disabled:opacity-50',
      secondary:
        'bg-gray-200 border border-gray-200 text-sm font-semibold text-gray-500 rounded-sm p-3.5 hover:not-disabled:border-blue-base disabled:opacity-50',
    },
  },
  defaultVariants: {
    variant: 'primary',
  },
});

type ButtonProps = ComponentProps<'button'> &
  VariantProps<typeof buttonVariants> & {
    icon?: ComponentType<IconProps>;
  };

export const Button: React.FC<ButtonProps> = ({
  variant,
  icon: Icon,
  children,
  ...props
}) => {
  return (
    <button type="button" className={buttonVariants({ variant })} {...props}>
      {Icon && (
        <Icon
          size="1rem"
          color="var(--color-gray-600)"
          data-testid="button-icon"
        />
      )}

      {children}
    </button>
  );
};
