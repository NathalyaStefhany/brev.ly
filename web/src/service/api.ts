import { env } from '@/env';
import axios from 'axios';

const baseURL = env.VITE_BACKEND_URL;

export const api = axios.create({
  baseURL,
});
