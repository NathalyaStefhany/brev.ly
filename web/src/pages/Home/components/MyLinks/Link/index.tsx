import React, { useState } from 'react';

import { Copy, Trash } from '@phosphor-icons/react';
import * as AlertDialog from '@radix-ui/react-alert-dialog';
import { IconButton } from '@/components/IconButton';
import { Button } from '@/components/Button';
import { api } from '@/service/api';
import { queryClient } from '@/service/queryClient';
import { Toast } from '@/components/Toast';

type LinkProps = {
  isFirstLink?: boolean;
  isLoading?: boolean;
  info?: {
    id: string;
    originalLink: string;
    shortenedLink: string;
    quantityAccesses: number;
  };
};

export const Link: React.FC<LinkProps> = ({
  isFirstLink = false,
  isLoading = false,
  info,
}) => {
  const [linkCopied, setLinkCopied] = useState(false);

  const [openDeleteDialog, setOpenDeleteDialog] = useState(false);
  const [isDeleting, setIsDeleting] = useState(false);
  const [deletionError, setDeletionError] = useState(false);

  const { id, originalLink, shortenedLink, quantityAccesses } = info ?? {};

  const handleCopyShortenedLink = (): void => {
    const link = `${window.location.origin}/${shortenedLink}`;

    navigator.clipboard.writeText(link);

    setLinkCopied(true);
  };

  const handleDeleteLink = async (): Promise<void> => {
    setIsDeleting(true);

    try {
      await api.delete(`/shortened-links/${shortenedLink}`);

      queryClient.refetchQueries({ queryKey: ['links list'], exact: true });

      setOpenDeleteDialog(false);
    } catch {
      setDeletionError(true);
    }

    setIsDeleting(false);
  };

  return (
    <>
      <div
        className="flex flex-row gap-8 md:gap-10 items-center border-t border-gray-200 pt-3 md:pt-4"
        style={{
          borderTopWidth: isFirstLink ? '0px' : '1px',
        }}
        data-testid={
          isLoading ? `container-loading-link` : `container-${id}-link`
        }
      >
        <div className="flex flex-col flex-1 gap-2 overflow-hidden">
          {isLoading ? (
            <div className="h-9 w-2/4 bg-gray-300 rounded animate-pulse" />
          ) : (
            <a
              href={`/${shortenedLink}`}
              target="_blank"
              rel="noreferrer"
              className="text-md text-blue-base truncate"
              data-testid="link-shortened-link"
            >
              brev.ly/{shortenedLink}
            </a>
          )}

          {isLoading ? (
            <div className="h-8 w-3/4 bg-gray-300 rounded animate-pulse" />
          ) : (
            <p
              className="text-sm text-gray-500 truncate"
              data-testid="text-original-link"
            >
              {originalLink}
            </p>
          )}
        </div>

        {isLoading ? (
          <div className="h-8 w-27 bg-gray-300 rounded animate-pulse" />
        ) : (
          <p
            className="text-sm text-gray-500 whitespace-nowrap"
            data-testid="text-quantity-accesses"
          >
            {quantityAccesses} acessos
          </p>
        )}

        <div className="flex flex-row gap-2 items-center">
          {isLoading ? (
            <>
              <div className="h-16 w-16 bg-gray-300 rounded animate-pulse" />{' '}
              <div className="h-16 w-16 bg-gray-300 rounded animate-pulse" />
            </>
          ) : (
            <>
              <IconButton
                icon={Copy}
                onClick={handleCopyShortenedLink}
                data-testid="button-copy"
              />

              <AlertDialog.Root
                open={openDeleteDialog}
                onOpenChange={setOpenDeleteDialog}
              >
                <AlertDialog.Overlay className="fixed inset-0 bg-gray-600/30 z-2" />

                <AlertDialog.Trigger asChild>
                  <IconButton icon={Trash} data-testid="button-delete" />
                </AlertDialog.Trigger>

                <AlertDialog.Content
                  className="w-150 z-3 fixed top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 bg-white shadow-lg shadow-gray-400 p-8 rounded-lg"
                  data-testid="container-delete-confirmation"
                >
                  <AlertDialog.Title className="text-lg text-gray-600 mb-4">
                    Apagar link
                  </AlertDialog.Title>

                  <AlertDialog.Description className="text-sm text-gray-600 mb-12">
                    Você realmente quer apagar o link {shortenedLink}?
                  </AlertDialog.Description>

                  <div className="flex flex-row gap-4 justify-end">
                    <AlertDialog.Cancel asChild>
                      <Button
                        variant="secondary"
                        style={{ padding: '0.5rem 1.25rem' }}
                        data-testid="button-cancel-deletion"
                      >
                        Cancelar
                      </Button>
                    </AlertDialog.Cancel>

                    <Button
                      onClick={handleDeleteLink}
                      isLoading={isDeleting}
                      disabled={isDeleting}
                      style={{
                        width: 'fit-content',
                        padding: '0.5rem 1.25rem',
                        borderRadius: '4px',
                      }}
                      data-testid="button-delete-confirmation"
                    >
                      Apagar
                    </Button>
                  </div>
                </AlertDialog.Content>
              </AlertDialog.Root>
            </>
          )}
        </div>
      </div>

      <Toast
        id="toast-shortened-link-copied"
        type="information"
        title="Link copiado com sucesso"
        description={`O link ${shortenedLink} foi copiado para a área de transferência`}
        duration={3000}
        open={linkCopied}
        onOpenChange={setLinkCopied}
      />

      <Toast
        id="toast-deletion-error"
        type="error"
        title="Erro ao deletar"
        description="Por favor, tente novamente mais tarde."
        open={deletionError}
        onOpenChange={setDeletionError}
      />
    </>
  );
};
