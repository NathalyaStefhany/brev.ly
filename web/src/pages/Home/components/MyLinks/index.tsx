import React from 'react';

import {
  DownloadSimple,
  LinkBreak,
  Link as LinkIcon,
} from '@phosphor-icons/react';
import { useInfiniteQuery } from '@tanstack/react-query';
import * as ScrollArea from '@radix-ui/react-scroll-area';
import { Button } from '@/components/Button';
import { api } from '@/service/api';
import { Spinner } from '@/components/Spinner';
import { Link } from '@/pages/Home/components/MyLinks/Link';
import { EmptyState } from '@/components/EmptyState';

const PAGE_SIZE = 20;

type ShortenedLinkInfo = {
  id: string;
  originalLink: string;
  shortenedLink: string;
  quantityAccesses: number;
  createdAt: string;
};

type ShortenedLinkList = {
  total: number;
  page: number;
  pageSize: number;
  data: ShortenedLinkInfo[];
};

export const MyLinks: React.FC = () => {
  const {
    data: listOfLinks,
    isError: listOfLinksIsError,
    isFetching: listOfLinksIsFetching,
    isRefetching: listOfLinksIsRefetching,
    hasNextPage,
    fetchNextPage,
  } = useInfiniteQuery<ShortenedLinkList>({
    queryKey: ['links list'],
    queryFn: async ({ pageParam }) => {
      const { data } = await api.get(
        `/shortened-links?page=${pageParam}&pageSize=${PAGE_SIZE}`,
      );

      return data;
    },
    initialPageParam: 1,
    getNextPageParam: (lastPage, allPages) => {
      if (lastPage.total > allPages.length * PAGE_SIZE) {
        return lastPage.page + 1;
      }

      return undefined;
    },
  });

  const onScroll = (event: React.UIEvent<HTMLDivElement>): void => {
    const { scrollHeight, scrollTop, clientHeight } = event.currentTarget;

    const distanceFromScrollToEnd = scrollHeight - scrollTop - clientHeight;
    const scrollIsMiddle = distanceFromScrollToEnd <= clientHeight / 2;

    if (scrollIsMiddle && hasNextPage && !listOfLinksIsFetching) {
      fetchNextPage();
    }
  };

  const quantityOfLinks = (listOfLinks && listOfLinks.pages[0].total) ?? 0;
  const listOfLinksIsEmpty = quantityOfLinks === 0;

  return (
    <div className="relative max-w-190 w-full md:max-w-290 md:min-w-190 h-fit flex flex-col flex-1 gap-8 md:gap-10 bg-gray-100 rounded-lg p-12 md:p-16">
      {listOfLinksIsRefetching && (
        <div className="absolute top-0 left-0 h-1 w-full animate-border bg-size-[100px_auto] md:bg-size-[200px_auto] bg-no-repeat bg-gradient-to-r from-blue-base to-blue-base" />
      )}

      <div className="flex flex-row items-center justify-between">
        <h2 className="text-lg text-gray-600">Meus links</h2>

        <Button variant="secondary" icon={DownloadSimple} disabled>
          Baixar CSV
        </Button>
      </div>

      <div className="border-t border-gray-200">
        {listOfLinksIsError ? (
          <EmptyState
            icon={<LinkBreak size="2rem" color="var(--color-danger)" />}
            description="Erro ao carregar os links cadastrados"
            data-testid="container-link-list-error"
          />
        ) : !listOfLinks ? (
          <EmptyState
            icon={<Spinner />}
            description="Carregando..."
            data-testid="container-link-list-loading"
          />
        ) : listOfLinksIsEmpty ? (
          <EmptyState
            icon={<LinkIcon size="2rem" color="var(--color-gray-400)" />}
            description="Ainda nÃ£o existem links cadastrados"
            data-testid="container-link-list-empty"
          />
        ) : (
          <ScrollArea.Root type="auto" className="w-full">
            <ScrollArea.Viewport
              onScroll={onScroll}
              className="overflow-hidden"
              style={{
                maxHeight: 'calc(100vh - 19rem)',
                minHeight: quantityOfLinks < 4 ? 'fit-content' : '14.125rem',
              }}
              data-testid="container-my-links-scroll"
            >
              <div className="md:w-full flex flex-col gap-3 md:gap-4">
                {listOfLinks.pages.map(({ page, data }) =>
                  data.map((link, index) => {
                    const isFirstLink = index === 0 && page === 1;

                    return (
                      <Link
                        key={link.id}
                        isFirstLink={isFirstLink}
                        info={link}
                      />
                    );
                  }),
                )}

                {hasNextPage && <Link isFirstLink={false} isLoading />}
              </div>
            </ScrollArea.Viewport>

            <ScrollArea.Scrollbar
              orientation="vertical"
              className="flex w-4 mr-[-0.875rem] bg-gray-200 rounded-full transition-colors duration-150 ease-out"
            >
              <ScrollArea.Thumb className="flex-1 rounded-full bg-blue-base hover:bg-blue-dark" />
            </ScrollArea.Scrollbar>
          </ScrollArea.Root>
        )}
      </div>
    </div>
  );
};
