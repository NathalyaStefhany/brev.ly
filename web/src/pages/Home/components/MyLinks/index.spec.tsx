import { beforeEach, describe, expect, it } from 'vitest';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { QueryClientProvider } from '@tanstack/react-query';
import AxiosMock from 'axios-mock-adapter';
import { MyLinks } from '@/pages/Home/components/MyLinks';
import { queryClient } from '@/service/queryClient';
import { api } from '@/service/api';

queryClient.setDefaultOptions({ queries: { retry: 0 } });

const apiMock = new AxiosMock(api);

describe('My Links testes', () => {
  beforeEach(() => {
    apiMock.reset();
    queryClient.clear();
  });

  it('should render correctly when list of links is empty', async () => {
    apiMock
      .onGet('/shortened-links?page=1&pageSize=20')
      .reply(200, { total: 0, page: 1, pageSize: 20, data: [] });

    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <MyLinks />
      </QueryClientProvider>,
    );

    await waitFor(() => {
      expect(
        apiMock.history.get.some(
          ({ url }) => url === '/shortened-links?page=1&pageSize=20',
        ),
      ).toBeTruthy();
    });

    await waitFor(() => {
      expect(
        screen.getByTestId('container-link-list-empty'),
      ).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it('should render correctly when it is loading list of links', async () => {
    apiMock
      .onGet('/shortened-links?page=1&pageSize=20')
      .reply(200, { total: 0, page: 1, pageSize: 20, data: [] });

    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <MyLinks />
      </QueryClientProvider>,
    );

    expect(
      screen.getByTestId('container-link-list-loading'),
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();

    await waitFor(() => {
      expect(
        screen.getByTestId('container-link-list-empty'),
      ).toBeInTheDocument();
    });
  });

  it('should render correctly when list of links request fails', async () => {
    apiMock.onGet('/shortened-links?page=1&pageSize=20').reply(400);

    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <MyLinks />
      </QueryClientProvider>,
    );

    await waitFor(() => {
      expect(
        apiMock.history.get.some(
          ({ url }) => url === '/shortened-links?page=1&pageSize=20',
        ),
      ).toBeTruthy();
    });

    await waitFor(() => {
      expect(
        screen.getByTestId('container-link-list-error'),
      ).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it('should display list of links correclty', async () => {
    apiMock.onGet('/shortened-links?page=1&pageSize=20').reply(200, {
      total: 10,
      page: 1,
      pageSize: 20,
      data: [...Array(10)].map((_, i) => ({
        id: i.toString(),
        shortenedLink: `test-${i}`,
        originalLink: `https://test-${i}.com`,
        quantityAccesses: i,
      })),
    });

    render(
      <QueryClientProvider client={queryClient}>
        <MyLinks />
      </QueryClientProvider>,
    );

    await waitFor(() => {
      expect(
        apiMock.history.get.some(
          ({ url }) => url === '/shortened-links?page=1&pageSize=20',
        ),
      ).toBeTruthy();
    });

    await waitFor(() => {
      expect(
        screen.queryByTestId('container-link-list-loading'),
      ).not.toBeInTheDocument();
    });

    for (let i = 0; i < 10; i++) {
      expect(screen.getByTestId(`container-${i}-link`)).toBeInTheDocument();

      expect(screen.getByTestId(`container-${i}-link`)).toHaveTextContent(
        `brev.ly/test-${i}`,
      );
      expect(screen.getByTestId(`container-${i}-link`)).toHaveTextContent(
        `https://test-${i}.com`,
      );
      expect(screen.getByTestId(`container-${i}-link`)).toHaveTextContent(
        `${i} acessos`,
      );
    }
  });

  it('should display skeleton loading in the end of list when has more pages to load', async () => {
    apiMock.onGet('/shortened-links?page=1&pageSize=20').reply(200, {
      total: 40,
      page: 1,
      pageSize: 20,
      data: [...Array(20)].map((_, i) => ({
        id: i.toString(),
        shortenedLink: `test-${i}`,
        originalLink: `https://test-${i}.com`,
        quantityAccesses: i,
      })),
    });

    render(
      <QueryClientProvider client={queryClient}>
        <MyLinks />
      </QueryClientProvider>,
    );

    await waitFor(() => {
      expect(
        apiMock.history.get.some(
          ({ url }) => url === '/shortened-links?page=1&pageSize=20',
        ),
      ).toBeTruthy();
    });

    await waitFor(() => {
      expect(
        screen.queryByTestId('container-link-list-loading'),
      ).not.toBeInTheDocument();
    });

    expect(screen.getByTestId('container-loading-link')).toBeInTheDocument();
  });

  it('should load the next page when go to the end of list', async () => {
    apiMock.onGet('/shortened-links?page=1&pageSize=20').reply(200, {
      total: 40,
      page: 1,
      pageSize: 20,
      data: [...Array(20)].map((_, i) => ({
        id: i.toString(),
        shortenedLink: `test-${i}`,
        originalLink: `https://test-${i}.com`,
        quantityAccesses: i,
      })),
    });

    apiMock.onGet('/shortened-links?page=2&pageSize=20').reply(200, {
      total: 40,
      page: 2,
      pageSize: 20,
      data: [...Array(20)].map((_, i) => {
        const index = i + 20;

        return {
          id: index.toString(),
          shortenedLink: `test-${index}`,
          originalLink: `https://test-${index}.com`,
          quantityAccesses: index,
        };
      }),
    });

    render(
      <QueryClientProvider client={queryClient}>
        <MyLinks />
      </QueryClientProvider>,
    );

    await waitFor(() => {
      expect(
        apiMock.history.get.some(
          ({ url }) => url === '/shortened-links?page=1&pageSize=20',
        ),
      ).toBeTruthy();
    });

    await waitFor(() => {
      expect(
        screen.queryByTestId('container-link-list-loading'),
      ).not.toBeInTheDocument();
    });

    for (let i = 0; i < 40; i++) {
      if (i < 20) {
        expect(screen.getByTestId(`container-${i}-link`)).toBeInTheDocument();

        expect(screen.getByTestId(`container-${i}-link`)).toHaveTextContent(
          `brev.ly/test-${i}`,
        );
        expect(screen.getByTestId(`container-${i}-link`)).toHaveTextContent(
          `https://test-${i}.com`,
        );
        expect(screen.getByTestId(`container-${i}-link`)).toHaveTextContent(
          `${i} acessos`,
        );
      } else {
        expect(
          screen.queryByTestId(`container-${i}-link`),
        ).not.toBeInTheDocument();
      }
    }

    fireEvent.scroll(screen.getByTestId('container-my-links-scroll'), {
      currentTarget: { scrollY: 200 },
    });

    await waitFor(() => {
      expect(
        apiMock.history.get.some(
          ({ url }) => url === '/shortened-links?page=2&pageSize=20',
        ),
      ).toBeTruthy();
    });

    for (let i = 0; i < 40; i++) {
      expect(screen.getByTestId(`container-${i}-link`)).toBeInTheDocument();

      expect(screen.getByTestId(`container-${i}-link`)).toHaveTextContent(
        `brev.ly/test-${i}`,
      );
      expect(screen.getByTestId(`container-${i}-link`)).toHaveTextContent(
        `https://test-${i}.com`,
      );
      expect(screen.getByTestId(`container-${i}-link`)).toHaveTextContent(
        `${i} acessos`,
      );
    }

    expect(
      screen.queryByTestId('container-loading-link'),
    ).not.toBeInTheDocument();
  });

  it('should get report URL when click to download CSV', async () => {
    apiMock.onGet('/shortened-links?page=1&pageSize=20').reply(200, {
      total: 3,
      page: 1,
      pageSize: 20,
      data: [...Array(3)].map((_, i) => ({
        id: i.toString(),
        shortenedLink: `test-${i}`,
        originalLink: `https://test-${i}.com`,
        quantityAccesses: i,
      })),
    });
    apiMock
      .onGet('/shortened-links/export')
      .reply(200, { reportUrl: 'https://test.com' });

    render(
      <QueryClientProvider client={queryClient}>
        <MyLinks />
      </QueryClientProvider>,
    );

    await waitFor(() => {
      expect(
        apiMock.history.get.some(
          ({ url }) => url === '/shortened-links?page=1&pageSize=20',
        ),
      ).toBeTruthy();
    });

    await waitFor(() => {
      expect(
        screen.queryByTestId('container-link-list-loading'),
      ).not.toBeInTheDocument();
    });

    expect(screen.getByTestId('button-download')).not.toBeDisabled();

    fireEvent.click(screen.getByTestId('button-download'));

    expect(screen.getByTestId('button-download')).toBeDisabled();
    expect(screen.getByTestId('button-loading')).toBeInTheDocument();

    await waitFor(() => {
      expect(
        apiMock.history.get.some(
          ({ url }) => url === '/shortened-links/export',
        ),
      ).toBeTruthy();
    });
  });
});
