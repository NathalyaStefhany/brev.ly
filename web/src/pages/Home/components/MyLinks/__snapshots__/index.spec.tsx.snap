// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`My Links testes > should render correctly when it is loading list of links 1`] = `
<div>
  <div
    class="relative max-w-190 w-full md:max-w-290 md:min-w-190 h-fit flex flex-col flex-1 gap-8 md:gap-10 bg-gray-100 rounded-lg p-12 md:p-16"
  >
    <div
      class="flex flex-row items-center justify-between"
    >
      <h2
        class="text-lg text-gray-600"
      >
        Meus links
      </h2>
      <button
        class="flex flex-row gap-3 items-center justify-center cursor-pointer transition-all duration-300 disabled:cursor-not-allowed bg-gray-200 border border-gray-200 text-sm font-semibold text-gray-500 rounded-sm p-3.5 hover:not-disabled:border-blue-base disabled:opacity-50"
        disabled=""
        type="button"
      >
        <svg
          data-testid="button-icon"
          fill="var(--color-gray-600)"
          height="1rem"
          viewBox="0 0 256 256"
          width="1rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224,144v64a8,8,0,0,1-8,8H40a8,8,0,0,1-8-8V144a8,8,0,0,1,16,0v56H208V144a8,8,0,0,1,16,0Zm-101.66,5.66a8,8,0,0,0,11.32,0l40-40a8,8,0,0,0-11.32-11.32L136,124.69V32a8,8,0,0,0-16,0v92.69L93.66,98.34a8,8,0,0,0-11.32,11.32Z"
          />
        </svg>
        Baixar CSV
      </button>
    </div>
    <div
      class="border-t border-gray-200"
    >
      <div
        class="flex flex-col items-center justify-center gap-6 mt-16 mb-12"
        data-testid="container-link-list-loading"
      >
        <div
          class="relative w-12 h-12"
        >
          <div
            class="absolute top-4.5 left-5.5 w-1 h-3 bg-blue-base rounded origin-center animate-fade"
            style="transform: rotate(0deg) translateY(-150%); opacity: 0; animation-delay: 0s;"
          />
          <div
            class="absolute top-4.5 left-5.5 w-1 h-3 bg-blue-base rounded origin-center animate-fade"
            style="transform: rotate(36deg) translateY(-150%); opacity: 0.1; animation-delay: 0.098s;"
          />
          <div
            class="absolute top-4.5 left-5.5 w-1 h-3 bg-blue-base rounded origin-center animate-fade"
            style="transform: rotate(72deg) translateY(-150%); opacity: 0.2; animation-delay: 0.196s;"
          />
          <div
            class="absolute top-4.5 left-5.5 w-1 h-3 bg-blue-base rounded origin-center animate-fade"
            style="transform: rotate(108deg) translateY(-150%); opacity: 0.3; animation-delay: 0.29400000000000004s;"
          />
          <div
            class="absolute top-4.5 left-5.5 w-1 h-3 bg-blue-base rounded origin-center animate-fade"
            style="transform: rotate(144deg) translateY(-150%); opacity: 0.4; animation-delay: 0.392s;"
          />
          <div
            class="absolute top-4.5 left-5.5 w-1 h-3 bg-blue-base rounded origin-center animate-fade"
            style="transform: rotate(180deg) translateY(-150%); opacity: 0.5; animation-delay: 0.49s;"
          />
          <div
            class="absolute top-4.5 left-5.5 w-1 h-3 bg-blue-base rounded origin-center animate-fade"
            style="transform: rotate(216deg) translateY(-150%); opacity: 0.6; animation-delay: 0.5880000000000001s;"
          />
          <div
            class="absolute top-4.5 left-5.5 w-1 h-3 bg-blue-base rounded origin-center animate-fade"
            style="transform: rotate(252deg) translateY(-150%); opacity: 0.7; animation-delay: 0.686s;"
          />
          <div
            class="absolute top-4.5 left-5.5 w-1 h-3 bg-blue-base rounded origin-center animate-fade"
            style="transform: rotate(288deg) translateY(-150%); opacity: 0.8; animation-delay: 0.784s;"
          />
          <div
            class="absolute top-4.5 left-5.5 w-1 h-3 bg-blue-base rounded origin-center animate-fade"
            style="transform: rotate(324deg) translateY(-150%); opacity: 0.9; animation-delay: 0.882s;"
          />
        </div>
        <p
          class="text-xs text-gray-500 uppercase text-center"
        >
          Carregando...
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`My Links testes > should render correctly when list of links is empty 1`] = `
<div>
  <div
    class="relative max-w-190 w-full md:max-w-290 md:min-w-190 h-fit flex flex-col flex-1 gap-8 md:gap-10 bg-gray-100 rounded-lg p-12 md:p-16"
  >
    <div
      class="flex flex-row items-center justify-between"
    >
      <h2
        class="text-lg text-gray-600"
      >
        Meus links
      </h2>
      <button
        class="flex flex-row gap-3 items-center justify-center cursor-pointer transition-all duration-300 disabled:cursor-not-allowed bg-gray-200 border border-gray-200 text-sm font-semibold text-gray-500 rounded-sm p-3.5 hover:not-disabled:border-blue-base disabled:opacity-50"
        disabled=""
        type="button"
      >
        <svg
          data-testid="button-icon"
          fill="var(--color-gray-600)"
          height="1rem"
          viewBox="0 0 256 256"
          width="1rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224,144v64a8,8,0,0,1-8,8H40a8,8,0,0,1-8-8V144a8,8,0,0,1,16,0v56H208V144a8,8,0,0,1,16,0Zm-101.66,5.66a8,8,0,0,0,11.32,0l40-40a8,8,0,0,0-11.32-11.32L136,124.69V32a8,8,0,0,0-16,0v92.69L93.66,98.34a8,8,0,0,0-11.32,11.32Z"
          />
        </svg>
        Baixar CSV
      </button>
    </div>
    <div
      class="border-t border-gray-200"
    >
      <div
        class="flex flex-col items-center justify-center gap-6 mt-16 mb-12"
        data-testid="container-link-list-empty"
      >
        <svg
          fill="var(--color-gray-400)"
          height="2rem"
          viewBox="0 0 256 256"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M240,88.23a54.43,54.43,0,0,1-16,37L189.25,160a54.27,54.27,0,0,1-38.63,16h-.05A54.63,54.63,0,0,1,96,119.84a8,8,0,0,1,16,.45A38.62,38.62,0,0,0,150.58,160h0a38.39,38.39,0,0,0,27.31-11.31l34.75-34.75a38.63,38.63,0,0,0-54.63-54.63l-11,11A8,8,0,0,1,135.7,59l11-11A54.65,54.65,0,0,1,224,48,54.86,54.86,0,0,1,240,88.23ZM109,185.66l-11,11A38.41,38.41,0,0,1,70.6,208h0a38.63,38.63,0,0,1-27.29-65.94L78,107.31A38.63,38.63,0,0,1,144,135.71a8,8,0,0,0,16,.45A54.86,54.86,0,0,0,144,96a54.65,54.65,0,0,0-77.27,0L32,130.75A54.62,54.62,0,0,0,70.56,224h0a54.28,54.28,0,0,0,38.64-16l11-11A8,8,0,0,0,109,185.66Z"
          />
        </svg>
        <p
          class="text-xs text-gray-500 uppercase text-center"
        >
          Ainda n√£o existem links cadastrados
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`My Links testes > should render correctly when list of links request fails 1`] = `
<div>
  <div
    class="relative max-w-190 w-full md:max-w-290 md:min-w-190 h-fit flex flex-col flex-1 gap-8 md:gap-10 bg-gray-100 rounded-lg p-12 md:p-16"
  >
    <div
      class="flex flex-row items-center justify-between"
    >
      <h2
        class="text-lg text-gray-600"
      >
        Meus links
      </h2>
      <button
        class="flex flex-row gap-3 items-center justify-center cursor-pointer transition-all duration-300 disabled:cursor-not-allowed bg-gray-200 border border-gray-200 text-sm font-semibold text-gray-500 rounded-sm p-3.5 hover:not-disabled:border-blue-base disabled:opacity-50"
        disabled=""
        type="button"
      >
        <svg
          data-testid="button-icon"
          fill="var(--color-gray-600)"
          height="1rem"
          viewBox="0 0 256 256"
          width="1rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224,144v64a8,8,0,0,1-8,8H40a8,8,0,0,1-8-8V144a8,8,0,0,1,16,0v56H208V144a8,8,0,0,1,16,0Zm-101.66,5.66a8,8,0,0,0,11.32,0l40-40a8,8,0,0,0-11.32-11.32L136,124.69V32a8,8,0,0,0-16,0v92.69L93.66,98.34a8,8,0,0,0-11.32,11.32Z"
          />
        </svg>
        Baixar CSV
      </button>
    </div>
    <div
      class="border-t border-gray-200"
    >
      <div
        class="flex flex-col items-center justify-center gap-6 mt-16 mb-12"
        data-testid="container-link-list-error"
      >
        <svg
          fill="var(--color-danger)"
          height="2rem"
          viewBox="0 0 256 256"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M198.63,57.37a32,32,0,0,0-45.19-.06L141.79,69.52a8,8,0,0,1-11.58-11l11.72-12.29a1.59,1.59,0,0,1,.13-.13,48,48,0,0,1,67.88,67.88,1.59,1.59,0,0,1-.13.13l-12.29,11.72a8,8,0,0,1-11-11.58l12.21-11.65A32,32,0,0,0,198.63,57.37ZM114.21,186.48l-11.65,12.21a32,32,0,0,1-45.25-45.25l12.21-11.65a8,8,0,0,0-11-11.58L46.19,141.93a1.59,1.59,0,0,0-.13.13,48,48,0,0,0,67.88,67.88,1.59,1.59,0,0,0,.13-.13l11.72-12.29a8,8,0,1,0-11.58-11ZM216,152H192a8,8,0,0,0,0,16h24a8,8,0,0,0,0-16ZM40,104H64a8,8,0,0,0,0-16H40a8,8,0,0,0,0,16Zm120,80a8,8,0,0,0-8,8v24a8,8,0,0,0,16,0V192A8,8,0,0,0,160,184ZM96,72a8,8,0,0,0,8-8V40a8,8,0,0,0-16,0V64A8,8,0,0,0,96,72Z"
          />
        </svg>
        <p
          class="text-xs text-gray-500 uppercase text-center"
        >
          Erro ao carregar os links cadastrados
        </p>
      </div>
    </div>
  </div>
</div>
`;
